src/twisted/test/test_udp.py

```diff
@@ -307,6 +307,66 @@ def cbFinished(ignored):
         d.addCallback(cbFinished)
         return d
 
+    def test_server_read_failure(self):
+        """
+        When a server fails to successfully read a packet the server should
+        still be able to process future packets.
+        The IOCP reactor had a historical problem where a failure to read caused
+        the reactor to ignore any future reads. This test should prevent a regression.
+
+        Note: This test assumes no one is listening on port 80 UDP.
```

Comment:
This should allocate a port in the safe way for tests (i.e.: listen on port zero and get the bound port).